nix/lib.nix

```diff
@@ -67,13 +67,14 @@
       # For master identies with no explicit pubkey, try extracting a pubkey from the file first.
       # Collect final identity arguments for encryption in an array.
       masterIdentityArgs=()
-      # shellcheck disable=SC2041
-      for file in ${
+      masterIdentityArray=(${
         concatStringsSep " "
         (map
-          (x: "${escapeShellArg x.identity}")
+          (x: ''"${escapeShellArg x.identity}"'')
```

Comment:
By adding an extra layer of quoting, this appears to have broken my attempts to create new secrets with a master identity present.

Prior to this change, this block of code would transform the nix array: ```
["/a/b" "/c/d"]
 -> [ "'/a/b'" "'/c/d'" ] # escapeShellArg
 -> [ "'/a/b'" "'/c/d'" ] # x: "${x}"
 -> "'/a/b' '/c/d'" # concatStringsSep
 # bash runtime
 -> ('/a/b' '/c/d') # bash array with two elements
```

Following this change, the following happens instead:

```
["/a/b" "/c/d"]
 -> [ "'/a/b'" "'/c/d'" ] # escapeShellArg
 -> [ ''"'/a/b'"'' ''"'/c/d'"'' ] # x: ''"${x}"''
 -> ''"'/a/b'" "'/c/d'"'' # concatStringsSep
 # bash runtime
 -> ("'/a/b'" "'/c/d'") # bash array with two elements
```

When these values attempt to get used later, this results grep, age, etc. attempting to open "the file named `/a/b` in the current working directory" instead of "the file named `b` in the directory `/a/`".

I /think/ this is as simple a fix as reverting line 73 here and dropping the extra layer of double quotes, but I don't have the larger context here.

These error messages really sucked to debug as a newbie. Didn't figure out what was going on until I pulled out strace and started digging through the `execve` calls.

Reply:
Changing my flake to check out this commit's parent:

```
inputs.agenix-rekey.url = "github:oddlama/agenix-rekey/126b4a5133eb361cbf5bf90e44c71b6f830845ec";
```

resolved the issue, so I'm pretty sure this change introduced the bug